src/Shouldly/MessageGenerators/DictionaryShouldContainKeyAndValueMessageGenerator.cs

```diff
@@ -6,11 +6,11 @@ namespace Shouldly.MessageGenerators;
 
 internal class DictionaryShouldContainKeyAndValueMessageGenerator : ShouldlyMessageGenerator
 {
-    private static readonly Regex Validator = new("ShouldContainKeyAndValue");
+    private static readonly Regex Validator = new Regex("ShouldContainKeyAndValue");
 
     public override bool CanProcess(IShouldlyAssertionContext context)
     {
-        return Validator.IsMatch(context.ShouldMethod);
+        return Validator.IsMatch(context.ShouldMethod) && context.Actual is IDictionary;
```

Comment:
We should support classes that implement `IReadOnlyDictionary<,>` and do *not* implement the non-generic `IDictionary`.